### PR TITLE
feat: Updated FAQ documentation

### LINKED ISSUE #950: Document how to implement translations
Likely to be not much more than a link to Godot's translation documentation. Should have some examples though - for both written and audio translations.

### LINKED ISSUE #990: Ability to swap out the sprites for the player (e.g change the player costume)-
**Is your feature request related to a problem? Please describe.**
I'd like to be able to change the player sprites so that I can make them change costume. Presumably the way would be to change the whole ESCRoom's player scene, but 

1) I can't see a supported way to do that.  It appears from the description that set_state might be intended to be used for that - "Can be used to change the appearance of an item or player character." - but I can't see code that matches that description.

2) The change needs to persist between rooms so needs to be a global rather than per-room setting.

**Describe the solution you'd like**
An easy way to set the sprites for the player. Ideally this solution would also be usable for ESCitems so that I don't need separate items for say a light that's on and a light that's off. 

**Describe alternatives you've considered**
None

**Additional context**

### LINKED ISSUE #1145: Menus, specific UIs, puzzles
- How to manage main menu, pause menu
- Some games propose specific UIs for the player, for puzzles or to add some realism in the environment (like an elevator's buttons panel)

### PR DESCRIPTION
**Pull request by balloonpopper**
Added documentation for changing character costume. Renamed the "other" documentation to FAQ and changed the headings to match.
Fixes https://github.com/godot-escoria/escoria-demo-game/issues/990
Fixes https://github.com/godot-escoria/escoria-demo-game/issues/950
Fixes https://github.com/godot-escoria/escoria-demo-game/issues/1145

### CODE REVIEW

File: advanced/faq.rst [hunk 1]
```diff
@@ -203,3 +201,95 @@ game is displayed is optional for your game. To create one:
 
    :newgame
    change_scene res://game/rooms/startroom/startroom.tscn
+
+
+How do I change the characters costume?
+---------------------------------------
+
+There are two ways to change the look of a character. The first is to create
+an entirely new character (i.e. a new Godot Scene with an ESCPlayer node,
```
Comment (BHSDuncan) [inline advanced/faq.rst]:
```suggestion
an entirely new character (i.e. a new Godot Scene with an `ESCPlayer` node,
```

File: advanced/faq.rst [hunk 2]
```diff
@@ -203,3 +201,95 @@ game is displayed is optional for your game. To create one:
...
+animated sprite, collision area, etc). The second option is to change just the
```
Comment (BHSDuncan) [inline advanced/faq.rst:211]:
```suggestion
animated sprite, collision area, etc). The second option is to change the
```

File: advanced/faq.rst [hunk 3]
```diff
@@ -203,3 +201,95 @@ game is displayed is optional for your game. To create one:
...
+animations (i.e. the sprites used) for the character.
+
+Changing the entire player scene
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This option might be appropriate if you needed to change something fundamental
```
Comment (BHSDuncan) [inline advanced/faq.rst]:
```suggestion
This option might be appropriate if you need to change something fundamental
```

File: advanced/faq.rst [hunk 4]
```diff
@@ -203,3 +201,95 @@ game is displayed is optional for your game. To create one:
...
+about a character for a particular scene (e.g. if for a specific level you only
+want the character to be able to walk in 2 directions where they normally have
+8 directions defined). If you choose this option, create the character scene,
+then in your game room (ESCRoom) set the PlayerScene parameter to point at the
```
Comment (BHSDuncan) [inline advanced/faq.rst]:
```suggestion
then in your game room (`ESCRoom`) set the `Player Scene` parameter to point at the
```

File: advanced/faq.rst [hunk 5]
```diff
@@ -203,3 +201,95 @@ game is displayed is optional for your game. To create one:
...
+new character scene.
+
+Changing the character animations
```
Comment (BHSDuncan) [inline advanced/faq.rst:224]:
```suggestion
Changing the character animations
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: advanced/faq.rst [hunk 6]
```diff
@@ -203,3 +201,95 @@ game is displayed is optional for your game. To create one:
...
+
+Changing the sprite set of a character so they look different (adding a hat,
+glasses, or changing their clothes for example) is an easy task. The process
```
Comment (BHSDuncan) [inline advanced/faq.rst]:
```suggestion
glasses, or changing their clothes, for example) is an easy task. The process
```

File: advanced/faq.rst [hunk 7]
```diff
@@ -203,3 +201,95 @@ game is displayed is optional for your game. To create one:
...
+involves creating new animations, then telling Escoria to change the character
+to use the animations as the default at the appropriate time.
```
Comment (BHSDuncan) [inline advanced/faq.rst]:
```suggestion
to use the animations as its default at the appropriate time.
```

File: advanced/faq.rst [hunk 8]
```diff
@@ -203,3 +201,95 @@ game is displayed is optional for your game. To create one:
...
+
+Create new animations
+^^^^^^^^^^^^^^^^^^^^^
+
+You should already have idle, talk and walk animations defined for your
```
Comment (BHSDuncan) [inline advanced/faq.rst]:
```suggestion
You should already have idle, talk, and walk animations defined for your
```

File: advanced/faq.rst [hunk 9]
```diff
@@ -203,3 +201,95 @@ game is displayed is optional for your game. To create one:
...
+   :alt: Animated sprite node
+
+Looking at the Animations window you should see the SpriteFrames defined.
+
+.. image:: img/character_animations_original.png
+   :alt: Defined animations
+
+Create new animations matching your new spriteset and call them something
+appropriate. In this example the demonstration character has had put on some
+jester clothes.
```
Comment (BHSDuncan) [inline advanced/faq.rst]:
```suggestion
appropriate. In this example, the demonstration character has had some
jester clothes put on.
```

File: advanced/faq.rst [hunk 10]
```diff
@@ -203,3 +201,95 @@ game is displayed is optional for your game. To create one:
...
+
+.. image:: img/character_animations_additional.png
+   :alt: Updated animations
+
+Go back to your character (ESCPlayer) node and using the dropdown arrow on the
```
Comment (BHSDuncan) [inline advanced/faq.rst]:
```suggestion
Go back to your character (`ESCPlayer`) node and, using the dropdown arrow on the
```

File: advanced/faq.rst [hunk 11]
```diff
@@ -203,3 +201,95 @@ game is displayed is optional for your game. To create one:
...
+
+As when you created your player originally, set up the correct number of
+directions for the character with the associated direction angles, as well as
+the direction, idles and speak animations pointing at your newly created
+animations.
+
+.. image:: img/character_animations_tres2.png
+   :alt: Updated animations
+
+Use the dropdown again choosing the **save** option. Give the file
```
Comment (BHSDuncan) [inline advanced/faq.rst]:
```suggestion
Use the dropdown again, choosing the **save** option. Give the file
```

File: advanced/faq.rst [hunk 12]
```diff
@@ -203,3 +201,95 @@ game is displayed is optional for your game. To create one:
...
+an appropriate name and location in the file dialogue window that appears.
```
Comment (BHSDuncan) [inline advanced/faq.rst]:
```suggestion
an appropriate name and location in the file dialog window that appears.
```

File: advanced/faq.rst [hunk 13]
```diff
@@ -203,3 +201,95 @@ game is displayed is optional for your game. To create one:
...
+
+Assign the new animations to the character
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Now that you have the animations defined, you need to tell Escoria when you
+want to use them. If the player buys a Jester outfit for example, you may
```
Comment (BHSDuncan) [inline advanced/faq.rst]:
```suggestion
want to use them. If the player buys a Jester outfit, for example, you may
```

File: advanced/faq.rst [hunk 14]
```diff
@@ -203,3 +201,95 @@ game is displayed is optional for your game. To create one:
...
+  set_animations player res://game/characters/mark/mark_animations_jester.tres
+
+
+How do I translate my game into other languages?
+------------------------------------------------
+
+Escoria takes advantage of Godot's built-in translation functionality for
+providing language support. Translations information is found in Godot's
+Project/Project Settings/Localization menu (text in Translations, audio in
+Remaps).
```
Comment (BHSDuncan) [inline advanced/faq.rst]:
```suggestion
providing language support. Translation information is found in Godot's
`Project/Project Settings/Localization` menu (text in `Translations`, audio in
`Remaps`).
```

**Review by BHSDuncan — CHANGES_REQUESTED** [review]

File: advanced/faq.rst (hunk 5 above)
Comment (balloonpopper) [reply advanced/faq.rst:224]: @BHSDuncan  I'll add this separately. Git won't let me accept this suggestion for unknown reasons.

**Review by StraToN — APPROVED** [review]

**Comment by StraToN** [comment]
Fixes https://github.com/godot-escoria/escoria-issues/issues/27

**Review by BHSDuncan — APPROVED** [review]

**Pull request merged**